queries.sql

```diff
@@ -23,3 +24,67 @@ SELECT * FROM animals WHERE name!='Gabumon';
 
 -- Find all animals with a weight between 10.4kg and 17.3kg (including the animals with the weights that equals precisely 10.4kg or 17.3kg)
 SELECT * FROM animals WHERE weight_kg >= 10.4 AND weight_kg <= 17.3;
+
+--Phase 2 update animals table
+
+--Inside a transaction update the animals table by setting the species column to unspecified. Verify that change was made.
+--Then roll back the change and verify that species columns went back to the state before transaction
+
+BEGIN;
+UPDATE animals SET species = 'unspecified';
+SELECT * FROM animals;
+ROLLBACK;
+
+-- Transactions to all animals with names ending with mon
+
+BEGIN;
+UPDATE animals SET species = 'digimon' WHERE name LIKE '%mon';
+
+--Update the animals table by setting the species column to pokemon for all animals that don't have species already set.
+
+BEGIN;
+UPDATE animals SET species = 'pokemon' WHERE species is NULL;
+-- Commit the transaction
+COMMIT TRANSACTION;
+SELECT * FROM animals;
+
+--delete all records from animals and rollback
+
+BEGIN;
+DELETE FROM animals;
+ROLLBACK;
+SELECT * FROM animals;
+
+--Inside transaction 0.1
+
+BEGIN;
+DELETE FROM animals WHERE date_of_birth > '2020-01-01';
+SAVEPOINT DOBorn;
+UPDATE animals SET weight_kg = weight_kg * -1;
+ROLLBACK TO DOBorn;
+UPDATE animals SET weight_kg = weight_kg * -1 WHERE weight_kg < 0;
+COMMIT WORK;
+SELECT * FROM animals;
+
+--Queries to Answer the questions in the requirements:
+
+-- How many animals are there?
+SELECT COUNT(*) FROM animals;
+-- How many animals have never tried to escape?
+SELECT COUNT(escape_attempts)
+FROM animals
+WHERE escape_attempts = 0;
+-- What is the average weight of animals?
+SELECT AVG(weight_kg)
+FROM animals;
+-- Who escapes the most, neutered or not neutered animals?
+SELECT
+neutered,MAX(escape_attempts)
+FROM animals
+GROUP BY neutered
+-- What is the minimum and maximum weight of each type of animal?
+SELECT MIN(weight_kg),MAX(weight_kg) FROM animals;
```

Comment:
- Awesome query!! 😀. However, the result of the query should show the `minimum` and `maximum` weight of each type(in this case, `species`) of animals. 

Here 👇 is an example of what the query could look like. 😉 

```
SELECT species, MIN(weight_kg), MAX(weight_kg) FROM animals
   GROUP BY species;
```